--- fs/exec.c.bak	2025-06-16 16:39:10.561839686 -0400
+++ fs/exec.c	2025-06-16 16:40:27.085730043 -0400
@@ -1931,11 +1931,6 @@
 	if (IS_ERR(filename))
 		return PTR_ERR(filename);
 
-#ifdef CONFIG_KSU_SUSFS_SUS_SU
-	if (likely(susfs_is_sus_su_hooks_enabled)) {
-		ksu_handle_execveat_sucompat(&fd, &filename, &argv, &envp, &flags);
-	}
-#endif
 
 	/*
 	 * We move the actual failure in case of RLIMIT_NPROC excess from
