--- fs/open.c.bak	2025-06-16 17:12:01.838276453 -0400
+++ fs/open.c	2025-06-16 17:12:38.314251890 -0400
@@ -436,11 +436,6 @@
 	unsigned int lookup_flags = LOOKUP_FOLLOW;
 	const struct cred *old_cred = NULL;
 
-#ifdef CONFIG_KSU_SUSFS_SUS_SU
-	if (likely(susfs_is_sus_su_hooks_enabled)) {
-		ksu_handle_faccessat(&dfd, &filename, &mode, NULL);
-	}
-#endif
 
 	if (mode & ~S_IRWXO)	/* where's F_OK, X_OK, W_OK, R_OK? */
 		return -EINVAL;
