--- kernel/sys.c.bak	2025-05-26 20:28:13.057611039 -0400
+++ kernel/sys.c	2025-05-26 20:29:32.660097762 -0400
@@ -1297,6 +1297,9 @@
 	}
 	return ret;
 }
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+extern void susfs_spoof_uname(struct new_utsname* tmp);
+#endif
 
 SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 {
@@ -1306,6 +1309,9 @@
 
 	down_read(&uts_sem);
 	memcpy(&tmp, utsname(), sizeof(tmp));
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+	susfs_spoof_uname(&tmp);
+#endif
 	up_read(&uts_sem);
 
 	rcu_read_lock();
